lib/tasks/hydranorth-solr.rake

```diff
@@ -17,37 +40,65 @@ namespace :hydranorth do
       RakeLogger.info "reindexed #{id} used #{used_time}"
     end
 
-    desc "update the index on all GenericFiles"
-    task update_generic_file_index: :environment do
-      GenericFile.all.each(&:update_index)
-    end
-
     desc "Index with a pairtree"
     task "index_pairtree", [:input] => :environment do |t, args|
       input = args[:input]
       RakeLogger.info "***********START index_pairtree***************"
       read_config
+      puts "reindex #{input}"
+      index_pairtree(input)
+      RakeLogger.info "***********FINISH index_pairtree**************"
+    end
+
+    desc "Complete Reindex"
+    task "reindex_all" => :environment do |t, args|
+      RakeLogger.info "***********START reindex *********************"
+      read_config
       start = Time.now
-      objects = find_objects(input)
-      objects.each do |o|
-        index_single(o)
-      end
+      count = index_all_objects
       finish = Time.now
-      used_time = finish - start
-      RakeLogger.info "Indexed #{objects.size} objects, used #{used_time}"
+      used_time = finish-start
+      RakeLogger.info "A Complete Reindex of #{count} objects, used #{used_time}"
       RakeLogger.info "***********FINISH index_pairtree**************"
     end
 
     def read_config
       rails_env = Rails.env
- 
+
       config = YAML.load_file("config/fedora.yml")
       @user = config[rails_env]['user']
       @password = config[rails_env]['password']
       @rest = config[rails_env]['url']
       @base_path = config[rails_env]['base_path']
     end
 
+    def index_all_objects
+       count = 0
+       [(0..9),('a'..'z')].map {|i| i.to_a}.flatten.each do |a|
+          [(0..9),('a'..'z')].map {|i| i.to_a}.flatten.each do |b|
+            pairtree = a.to_s + b.to_s
+            number_reindexed = index_pairtree(pairtree)
+            count = count + number_reindexed
+         end
+       end
+       return count
+    end
+
+    def index_pairtree(pairtree)
+      RakeLogger.info "Start to reindex all objects starting with #{pairtree}"
+      puts "Start to reindex all objects starting with #{pairtree}"
```

Comment:
printing can probably be removed - it was simply there for the easy monitoring of the task run - but I can tail the log for the same info.